libs/velo-external-db-core/src/service/schema.ts

```diff
@@ -61,12 +63,135 @@ export default class SchemaService {
         }))
     }
 
-    
     async validateOperation(operationName: SchemaOperations) {
         const allowedSchemaOperations = this.storage.supportedOperations()
 
         if (!allowedSchemaOperations.includes(operationName)) 
             throw new errors.UnsupportedOperation(`Your database doesn't support ${operationName} operation`)
     }
 
+    async listCollections(collectionIds: string[]): Promise<ListCollectionsResponsePart> {
+        
+        // remove in the end of development
+        if (!this.storage.capabilities || !this.storage.columnCapabilitiesFor) {
+            throw new Error('Your storage does not support the new collection capabilities API')
+        }
+        
+        const collections = (!collectionIds || collectionIds.length === 0) ? 
+            await this.storage.list() : 
+            await Promise.all(collectionIds.map(async(collectionName: string) => ({ id: collectionName, fields: await this.schemaInformation.schemaFieldsFor(collectionName) })))
+                
+        const capabilities = this.formatCollectionCapabilities(this.storage.capabilities())
+        const collectionAfterFormat: Collection[] = collections.map((collection) => ({
+            id: collection.id,
+            fields: this.formatFields(collection.fields),
+            capabilities
+        }))
+
+        return { collection: collectionAfterFormat }
+    }
+
+    formatFields(fields: ResponseField[]): Field[] {
+        const getFieldCapabilities = (type: string): FieldCapabilities => {
+            // remove in the end of development
+            if (!this.storage.columnCapabilitiesFor) {
+                throw new Error('Your storage does not support the new collection capabilities API')
+            }
+            const { sortable, columnQueryOperators } = this.storage.columnCapabilitiesFor(type)
+            return {
+                sortable,
+                queryOperators: convertQueriesToQueryOperatorsEnum(columnQueryOperators)
+            }
+        }
+
+        return fields.map((field) => ({
+            key: field.field,
+            // TODO: think about how to implement this
+            encrypted: false,
+            type: convertFieldTypeToEnum(field.type),
+            capabilities: getFieldCapabilities(field.type)
+        }))
+    }
+
+    formatCollectionCapabilities(capabilities: DbCapabilities): CollectionCapabilities {
+        return {
+            dataOperations: capabilities.dataOperations as unknown as DataOperation[],
+            fieldTypes: capabilities.fieldTypes as unknown as FieldType[],
+            collectionOperations: capabilities.collectionOperations as unknown as CollectionOperation[],
```

Comment:
maybe create enums to `dataOperations`, `fieldTypes`, `collectionOperations` for the domain (even if it would be the same as the SPI), it would be more accurate than string and in these lines it would be more readable.

Reply:
This is what prepares to be in the next PR ...